### PR TITLE
Find escape by keysym not keycode

### PR DESCRIPTION
**Pull request by pickfire**
Keycode are hard representation of what key was pressed but it may not
be escape keysym.

### CODE REVIEW

**Comment by 9ary** [comment]
Looks good to me, this should indeed have been done from the beginning.
Minor nit: if the layout changes in the middle of a selection, the grab would become incorrect again. To fix that would require handling keymap changes, which is probably not worth it but whatever.

**Comment by tnjd** [comment]
Looks fine to me. Just downloaded and ran it and still works on my machine. Who needs automated testing, right?

File: src/lib/mod.rs [hunk 1]
```diff
@@ -139,23 +139,41 @@ pub fn grab_pointer_set_cursor(conn: &xcb::Connection, root: u32) -> bool {
...
+        setup.max_keycode() - setup.min_keycode() + 1,
+    );
+    let reply = cookie.get_reply().expect("failed to get keyboard mapping");
+
+    let escape_index = reply
+        .keysyms()
+        .iter()
+        .position(|&keysym| keysym == ESC_KEYSYM)
+        .expect("failed to find escape keysym");
+    return (escape_index / reply.keysyms_per_keycode() as usize) as u8 + setup.min_keycode();
```
Comment (tnjd) [inline src/lib/mod.rs]: minor nit: this 'return' is unneeded since it comes at the end of the function.

Comment (pickfire) [reply src/lib/mod.rs]: Oh, I forgot about that. Thanks for reminding, I am surprised clippy didn't mention this.

**Comment by tnjd** [comment]
Thanks!

**Pull request merged**